hack/git_client_side_hooks/pre-push

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+# Copyright 2022 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+COUNT_DOC_DIFF=$(git diff --name-only --staged '*.md' | wc -l)
```

Comment:
if you are going to check staged content, I suppose it can be a pre-commit hook, not a pre-push hook.

Reply:
Should we convert it into a pre-commit hook then?

Reply:
I feel it's Ok to change it as pre-commit hook, @antoninbas  any suggestion? thanks.

Reply:
My local git clone includes the following (sorry for the verbosity...):
```
$ cat .git/hooks/pre-push.sample
#!/bin/sh

# An example hook script to verify what is about to be pushed.  Called by "git
# push" after it has checked the remote status, but before anything has been
# pushed.  If this script exits with a non-zero status nothing will be pushed.
#
# This hook is called with the following parameters:
#
# $1 -- Name of the remote to which the push is being done
# $2 -- URL to which the push is being done
#
# If pushing without using a named remote those arguments will be equal.
#
# Information about the commits which are being pushed is supplied as lines to
# the standard input in the form:
#
#   <local ref> <local sha1> <remote ref> <remote sha1>
#
# This sample shows how to prevent push of commits where the log message starts
# with "WIP" (work in progress).

remote="$1"
url="$2"

z40=0000000000000000000000000000000000000000

while read local_ref local_sha remote_ref remote_sha
do
	if [ "$local_sha" = $z40 ]
	then
		# Handle delete
		:
	else
		if [ "$remote_sha" = $z40 ]
		then
			# New branch, examine all commits
			range="$local_sha"
		else
			# Update to existing branch, examine new commits
			range="$remote_sha..$local_sha"
		fi

		# Check for WIP commit
		commit=`git rev-list -n 1 --grep '^WIP' "$range"`
		if [ -n "$commit" ]
		then
			echo >&2 "Found WIP commit in $local_ref, not pushing"
			exit 1
		fi
	fi
done

exit 0
```
could we use these variables and keep this as a pre-push hook?
```
local_ref local_sha remote_ref remote_sha
```
I just feel like it could be a bit cumbersome as a pre-commit hook given that these commands (especially `make golangci` can take a bit of time to run).

Reply:
But even in that scenario, we'd still have to scan all the commits and run `make golangci` and `make verify` so in any case, it will take the same amount of time. In fact, it will take more time as it might be running these commands against several commits in case of multiple unpushed commits.

Reply:
I use `git commit -a --amend` a lot, and I may not want to run the checks all the time

I was thinking the following:
```
		if [ "$remote_sha" = $z40 ]
		then
			# New branch, run all tests
			make verify
                         make golangci
		else
			# Update to existing branch, examine new commits and decide whether we need to run checks
			diff=...
                         # run make verify if needed based on diff
                         # run make golangci if needed base on diff
		fi
```

With the above, all the checks will run for the first push, but afterwards only needed checks will run

That being said, if you don't believe this is a good solution, we can go with a post-commit hook, and see what the experience is like for developers after some time.